types/canonical.go

```diff
@@ -3,15 +3,65 @@ package types
 import (
 	"time"
 
+	cmtbytes "github.com/cometbft/cometbft/libs/bytes"
 	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
 	cmttime "github.com/cometbft/cometbft/types/time"
+	cmn "github.com/tendermint/tendermint/libs/common" //nolint:depguard
 )
 
 // Canonical* wraps the structs in types for amino encoding them for use in SignBytes / the Signable interface.
 
 // TimeFormat is used for generating the sigs
 const TimeFormat = time.RFC3339Nano
 
+// TODO(@raneet10): Check whether we should keep it
+type CanonicalBlockID struct {
+	Hash        cmtbytes.HexBytes
+	PartsHeader CanonicalPartSetHeader
+}
+
+// TODO(@raneet10): Check whether we should keep it
+type CanonicalPartSetHeader struct {
+	Hash  cmtbytes.HexBytes
+	Total uint32
+}
+
+// TODO(@raneet10): Check whether removing Data field affects serialization
+type CanonicalProposal struct {
+	Type      SignedMsgType // type alias for byte
+	Height    int64         `binary:"fixed64"`
+	Round     int64         `binary:"fixed64"`
+	POLRound  int64         `binary:"fixed64"`
+	BlockID   CanonicalBlockID
+	Timestamp time.Time
+	ChainID   string
+	Data      cmn.HexBytes // [peppermint] add data in proposal to that signers can sign
+}
+
+// TODO(@raneet10): Check whether removing Data and SideTxResults fields affect serialization
+type CanonicalVote struct {
+	Type      SignedMsgType // type alias for byte
+	Height    int64         `binary:"fixed64"`
+	Round     int64         `binary:"fixed64"`
+	BlockID   CanonicalBlockID
+	Timestamp time.Time
+	ChainID   string
+
+	Data          []byte         // [peppermint] tx hash
+	SideTxResults []SideTxResult // [peppermint] side tx results
+}
+
+// TODO(@raneet10): Check whether removing this struct affects serialization
+// CanonicalRLPVote create RLP vote to decode in contract
+// [peppermint]
+type CanonicalRLPVote struct {
```

Comment:
This one was also unused, even in the old Peppermint fork.
* If you don't find it used elsewhere in your codebase (SDK-fork, Heimdall-v2, etc.) just remove it
* If you find a usage of this, then move this to the repo where it is used

Reply:
Resolved